addons/account/tests/test_account_payment_duplicate.py

```diff
@@ -0,0 +1,166 @@
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestAccountPaymentDuplicateMoves(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        cls.receivable = cls.company_data['default_account_receivable']
+        cls.payable = cls.company_data['default_account_payable']
+        cls.bank_journal = cls.company_data['default_journal_bank']
+        cls.payment_in = cls.env['account.payment'].create({
+            'amount': 50.0,
+            'payment_type': 'inbound',
+            'partner_id': cls.partner_a.id,
+            'destination_account_id': cls.receivable.id,
+        })
+        cls.payment_out = cls.env['account.payment'].create({
+            'amount': 50.0,
+            'payment_type': 'outbound',
+            'partner_id': cls.partner_a.id,
+            'destination_account_id': cls.payable.id,
+        })
+
+    def test_duplicate_payments(self):
+        """
+            Ensure duplicated payments are computed correctly for both inbound and outbound payments.
+            For it to be a duplicate, the partner, the date and the amount must be the same.
+        """
+        payment_in_1 = self.payment_in
+        payment_out_1 = self.payment_out
+
+        # Different type but same partner, amount and date, no duplicate
+        self.assertRecordValues(payment_in_1, [{'duplicate_move_ids': []}])
+
+        # Create duplicate payments
+        payment_in_2 = payment_in_1.copy(default={'date': payment_in_1.date})
+        payment_out_2 = payment_out_1.copy(default={'date': payment_out_1.date})
+        # Inbound payment finds duplicate inbound payment, not the outbound payment with same information
+        self.assertRecordValues(payment_in_2, [{
+            'duplicate_move_ids': [payment_in_1.move_id.id]
```

Comment:
comma at the end for future iteration on the test, let's avoid breaking the github history (same for the others)